clash-lib/src/Clash/Netlist.hs

```diff
@@ -572,6 +570,7 @@ mkFunApp dstId fun args tickDecls = do
           (_,_,_,Component compName compInps co _) <- preserveVarEnv $ genComponent fun
           let argTys = map (termType tcm) args
           argHWTys <- mapM coreTypeToHWTypeM' argTys
+          voidDecls <- renderVoids (zip argHWTys args)
```

Comment:
Can't we just filter voids later so that `argDecls` contains the these voidDecls?

Reply:
Hmm, well, we'd later need to filter them again for the `length tysFiltered == length compInps` comparison and the `InstDecl` later on (specifically `inpAssigns`). So I'm not convinced this would improve readability that much.

Reply:
You'd pull the

```
(argExprs,argDecls)   <- fmap (second concat . unzip) $! mapM (\(e,t) -> mkExpr False (Left dstId) t e) argsFiltered'
```

out of the if-expression, and filter later. You'd still be filtering only once.

Reply:
Ah right, makes sense.